libs/template/writer.go

```diff
@@ -0,0 +1,166 @@
+package template
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/dbr"
+	"github.com/databricks/cli/libs/filer"
+)
+
+const (
+	libraryDirName  = "library"
+	templateDirName = "template"
+	schemaFileName  = "databricks_template_schema.json"
+)
+
+type Writer interface {
+	Configure(ctx context.Context, configPath, outputDir string) error
+	Materialize(ctx context.Context, r Reader) error
+	LogTelemetry(ctx context.Context) error
```

Comment:
This function does nothing right now. Telemetry functionality will be added in a followup PR.

Reply:
I recommend leaving this out to focus on the refactor of existing functionality.

It makes reviewing whether or not the structure makes sense harder because it's not yet used.